src/context/ListContext.tsx

```diff
@@ -0,0 +1,201 @@
+import {
+  FC, createContext, useState, useEffect, ReactNode,
+} from 'react';
+import {
+  User, Post, Comment, CommentData,
+} from '../types';
+import { getUsers } from '../api/users';
+import { getPosts } from '../api/posts';
+import { deleteComment, getComments, postComment } from '../api/comments';
+
+type Props = {
+  children: ReactNode;
+};
+
+interface IGlobalContext {
+  error: boolean;
+  resetError: () => void;
+}
+
+interface IPostsContext {
+  posts: Post[];
+  isPostsLoading: boolean;
+  selectedPost: Post | null,
+  onSelectPost: (post: Post) => void;
+}
+
+interface IUsersContext {
+  users: User[];
+  selectedUser: User | null;
+  isUsersLoading: boolean;
+  onSelectUser: (user: User) => void;
+}
+
+interface ICommentsContext {
+  comments: Comment[];
+  isCommentsLoading: boolean;
+  commentIsProcessing: boolean;
+  onAddNewComment: (comment: CommentData) => Promise<boolean>;
+  onDeleteComment: (commentId: number) => void;
+}
+
+export const GlobalContext = createContext({} as IGlobalContext);
+export const PostsContext = createContext({} as IPostsContext);
+export const UsersContext = createContext({} as IUsersContext);
+export const CommentsContext = createContext({} as ICommentsContext);
+
+export const ListProvider: FC<Props> = ({ children }) => {
+  const [error, setError] = useState(false);
+
+  // #region User states
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isUsersLoading, setIsUsersLoading] = useState(false);
+  // #endregion
+
+  // #region Posts states
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  // #endregion
+
+  // #region Comments states
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [commentIsProcessing, setCommentIsProcessing] = useState(false);
+  // #endregion
+
+  // #region load data
+
+  /* load users */
+  useEffect(() => {
+    setError(false);
+    setIsUsersLoading(true);
+
+    getUsers()
+      .then((usersList) => setUsers(usersList))
+      .catch(() => setError(true))
+      .finally(() => setIsUsersLoading(false));
+  }, []);
+
+  /* load posts for selected user */
+  useEffect(() => {
+    setError(false);
+    setSelectedPost(null);
```

Comment:
You forgot to set isPostsLoading to true before making the asynchronous request to get the posts.